lms/djangoapps/discussion_api/tests/test_views.py

```diff
@@ -154,6 +157,84 @@ def test_basic(self):
+    """Tests for CourseView"""
+    def setUp(self):
+        super(RetireViewTest, self).setUp()
+        self.superuser = SuperuserFactory()
+        self.hashed_username = get_retired_username_by_username(self.user.username)
+        self.url = reverse(
+            "retire_discussion_user",
+            kwargs={"username": text_type(self.user.username)}
+        )
+
+    def assert_response_correct(self, response, expected_status, expected_content):
+        """
+        Assert that the response has the given status code and content
+        """
+        self.assertEqual(response.status_code, expected_status)
+
+        if expected_content:
+            self.assertEqual(text_type(response.content), expected_content)
+
+    def build_jwt_headers(self, user):
+        """
+        Helper function for creating headers for the JWT authentication.
+        """
+        token = JwtBuilder(user).build_token([])
+        headers = {'HTTP_AUTHORIZATION': 'JWT ' + token}
+        return headers
+
+    def test_basic(self):
+        """
+        Check successful retirement case
+        """
+        self.register_get_user_retire_response(self.user)
+        headers = self.build_jwt_headers(self.superuser)
+        response = self.client.post(self.url, {'hashed_username': self.hashed_username}, **headers)
+        self.assert_response_correct(response, 204, "")
+
+    def test_bad_hash(self):
+        """
+        Check that we fail on a hash mismatch with an appropriate error
+        """
+        headers = self.build_jwt_headers(self.superuser)
+        response = self.client.post(self.url, {'hashed_username': "this will never match"}, **headers)
+        self.assert_response_correct(response, 500, '"Mismatched hashed_username, bad salt?"')
+
+    def test_downstream_forums_error(self):
+        """
+        Check that we bubble up errors from the comments service
+        """
+        self.register_get_user_retire_response(self.user, status=500, body="Server error")
+        headers = self.build_jwt_headers(self.superuser)
+        response = self.client.post(self.url, {'hashed_username': self.hashed_username}, **headers)
+        self.assert_response_correct(response, 500, '"Server error"')
+
+    def test_downstream_forums_error(self):
+        """
+        Check that we handle unknown users appropriately
+        """
+        nonexistent_username = "nonexistent user"
+        self.url = reverse(
+            "retire_discussion_user",
+            kwargs={"username": nonexistent_username}
+        )
+        self.hashed_username = get_retired_username_by_username(nonexistent_username)
+
+        headers = self.build_jwt_headers(self.superuser)
+        response = self.client.post(self.url, {'hashed_username': self.hashed_username}, **headers)
+        self.assert_response_correct(response, 404, None)
+
+    def test_not_authenticated(self):
+        """
+        Override the parent implementation of this, we auth differently
+        """
```

Comment:
Our call is authing the deletion user using JWT instead of the user performing the operation themselves, so that base method call fails for this test case.